x-pack/plugins/ml/public/register_feature.ts

```diff
@@ -28,7 +28,22 @@ export const registerFeature = (home: HomePublicPluginSetup) => {
     }),
     icon: 'machineLearningApp',
     path: '/app/ml',
+    showOnHomePage: false,
+    category: FeatureCatalogueCategory.DATA,
+  });
+
+  home.featureCatalogue.register({
+    id: `${PLUGIN_ID}_file_data_visualizer`,
+    title: i18n.translate('xpack.ml.fileDataVisualizerTitle', {
+      defaultMessage: 'Upload a file',
+    }),
+    description: i18n.translate('xpack.ml.fileDataVisualizerDescription', {
+      defaultMessage: 'Import your own CSV, NDJSON, or log file.',
+    }),
+    icon: 'importAction',
+    path: '/app/ml#/filedatavisualizer',
     showOnHomePage: true,
     category: FeatureCatalogueCategory.DATA,
+    order: 520,
```

Comment:
> ML changes LGTM
> Out of interest, what controls whether the "Upload a file" section appears?

@jgowdyelastic I registered the file upload card as an independent feature in the Feature Catalogue. Instead of checking the `canFindFileStructure` permission, I added `catalogue` privileges in `/ml/common/types/capabilities` which control access to entries in the Feature Catalogue service. I provide both the feature IDs for ML and the file upload feature in the privileges for the admin user catalogue privileges while the read only user only gets access to the ML card. 

https://github.com/elastic/kibana/pull/70571/files#diff-b8ea033359a430807fba7a93468fbf9f